src/webui/components/relation/elementary/FollowButton.tsx

```diff
@@ -1,16 +1,41 @@
 
+import { useState } from 'react';
 import { Button } from '../../../designsystem/module';
 
 export type Props = {
     isFollowing: boolean;
-    followHandler: () => void;
+    followHandler: () => Promise<void>;
 };
 
+type States = 'establish' | 'establishing' | 'established';
```

Comment:
States are supposed the have nouns as names. The term 'establish' is a verb, thus is not a valid state name. My suggestion is to use 'unestablished'.